debian/marquee/nginx/conf/whatwg.conf

```diff
@@ -0,0 +1,29 @@
+ssl_session_cache shared:SSL:10m;
+ssl_session_timeout 5m;
+
+ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+
+ssl_prefer_server_ciphers on;
+ssl_ciphers "ECDHE-RSA-AES128-GCM-SHA256 EECDH+ECDSA+AESGCM EECDH+aRSA+AESGCM EECDH+ECDSA+SHA384 EECDH+ECDSA+SHA256 EECDH+aRSA+SHA384 EECDH+aRSA+SHA256 EECDH+aRSA+RC4 EECDH EDH+aRSA RC4 !aNULL !eNULL !LOW !MD5 !EXP !PSK !SRP !DSS !MEDIUM !RC4 !CAMELLIA !3DES";
+
+ssl_dhparam /etc/ssl/certs/dhparam.pem;
+
+gzip on;
+gzip_disable "msie6";
+
+gzip_vary on;
+gzip_proxied any;
+gzip_comp_level 6;
+gzip_buffers 16 8k;
+gzip_http_version 1.1;
+gzip_types text/plain text/css application/json application/x-javascript text/xml application/xml image/svg+xml application/xml+rss text/javascript;
+
+include whatwg-headers.conf;
+
+location ~* \.(?:ico|css|js|gif|jpe?g|png|svg)$ {
+    # Some basic cache-control for static files to be sent to the browser
+    expires max;
+    add_header Pragma public;
+    add_header Cache-Control "public, proxy-revalidate";
+    include whatwg-headers.conf;
```

Comment:
include whatwg-headers.conf;

> Why is this included twice,

It’s necessary because of the way nginx handles config parsing

> won't the outer one suffice?

No, it won’t. See http://nginx.org/en/docs/http/ngx_http_headers_module.html#add_header:

>  There could be several `add_header` directives. These directives are inherited from the previous level if and only if there are no `add_header` directives defined on the current level.

So to be clear: because we have the  `add_header Cache-Control` and `add_header Pragma` directives defined in this `location` block (because we don’t want them included in all locations), any headers defined anywhere above that aren’t inherited. That’s exactly why I put those always-send-these headers in a separate file—so we can re-include them in other `location` blocks when needed.